Spigot-Server-Patches/0551-Optimize-Villagers.patch

```diff
@@ -121,51 +156,44 @@ index a6d8ef5eb44f3f851a3a1be4032ca21ab1d7f2b2..c3e8a0145d63843736d2060f978cdf38
  
 +    protected final Optional<R> getSection(long i) { return d(i); } // Paper - OBFHELPER
      protected Optional<R> d(long i) {
--        SectionPosition sectionposition = SectionPosition.a(i);
--
--        if (this.b(sectionposition)) {
--            return Optional.empty();
--        } else {
--            Optional<R> optional = this.c(i);
--
--            if (optional != null) {
--                return optional;
--            } else {
--                this.b(sectionposition.u());
--                optional = this.c(i);
--                if (optional == null) {
--                    throw (IllegalStateException) SystemUtils.c(new IllegalStateException());
--                } else {
--                    return optional;
--                }
--            }
--        }
-+        // Paper start - replace method - never load POI data sync, we load this in chunk load already, reduce ops
++        /* // Paper start - replaced logic
+         SectionPosition sectionposition = SectionPosition.a(i);
+ 
+         if (this.b(sectionposition)) {
+@@ -75,6 +77,11 @@ public class RegionFileSection<R> extends RegionFileCache implements AutoCloseab
+                 }
+             }
+         }
++         */
++        // Never load POI data sync, we load this in chunk load already, reduce ops
 +        // If it's an unloaded chunk, well too bad.
-+        return this.c(i);
++        return c(i);
 +        // Paper end
      }
  
      protected boolean b(SectionPosition sectionposition) {
-@@ -117,7 +103,7 @@ public class RegionFileSection<R extends MinecraftSerializable> extends RegionFi
+@@ -117,9 +124,11 @@ public class RegionFileSection<R> extends RegionFileCache implements AutoCloseab
+ 
      private <T> void a(ChunkCoordIntPair chunkcoordintpair, DynamicOps<T> dynamicops, @Nullable T t0) {
          if (t0 == null) {
++            /* // Paper start - NO!!!
              for (int i = 0; i < 16; ++i) {
--                this.c.put(SectionPosition.a(chunkcoordintpair, i).v(), Optional.empty());
-+                //this.c.put(SectionPosition.a(chunkcoordintpair, i).v(), Optional.empty()); // Paper - NO!!!
+                 this.c.put(SectionPosition.a(chunkcoordintpair, i).s(), Optional.empty());
              }
++             */ // Paper end
```

Comment:
no

Reply:
Why waste cpu time on even initializing the loop?

Reply:
The compiler optimized it out